docs/source/topics/cluster-setup.rst

```diff
@@ -0,0 +1,158 @@
+
+
+Configuring HBase
+=================
+* create a namespace ``crawler`` (see :setting:`HBASE_NAMESPACE`),
+* make sure Snappy compression is supported natively.
+
+
+Configuring Frontera
+====================
+Every Frontera component requires it's own configuration module, but some options are shared, so we recommend to create
+a common modules and import settings from it in component's modules.
+
+1. Create a common module and add there: ::
+
+    from __future__ import absolute_import
+    from frontera.settings.default_settings import MIDDLEWARES
+    MAX_NEXT_REQUESTS = 512
+    SPIDER_FEED_PARTITIONS = 2 # number of spider processes
+    SPIDER_LOG_PARTITIONS = 2 # worker instances
+    MIDDLEWARES.extend([
+        'frontera.contrib.middlewares.domain.DomainMiddleware',
+        'frontera.contrib.middlewares.fingerprint.DomainFingerprintMiddleware'
+    ])
+
+    QUEUE_HOSTNAME_PARTITIONING = True
+    KAFKA_LOCATION = 'localhost:9092' # your Kafka broker host:port
+    SCORING_TOPIC = 'frontier-scoring'
+    URL_FINGERPRINT_FUNCTION='frontera.utils.fingerprint.hostname_local_fingerprint'
+
+2. Create workers shared module: ::
+
+    from __future__ import absolute_import
+    from .common import *
+
+    BACKEND = 'frontera.contrib.backends.hbase.HBaseBackend'
+
+    MAX_NEXT_REQUESTS = 2048
+    NEW_BATCH_DELAY = 3.0
+
+    HBASE_THRIFT_HOST = 'localhost' # HBase Thrift server host and port
+    HBASE_THRIFT_PORT = 9090
+
+3. Create DB worker module: ::
+
+    from __future__ import absolute_import
+    from .worker import *
+
+    LOGGING_CONFIG='logging-db.conf' # if needed
+
+4. Create Strategy worker's module: ::
+
+    from __future__ import absolute_import
+    from .worker import *
+
+    CRAWLING_STRATEGY = '' # path to the crawling strategy class
+    LOGGING_CONFIG='logging-sw.conf' # if needed
+
+The logging can be configured according to https://docs.python.org/2/library/logging.config.html see the
+:doc:`list of loggers <loggers>`.
+
+5. Configure spiders module: ::
+
+    from __future__ import absolute_import
+    from .common import *
+
+    BACKEND = 'frontera.contrib.backends.remote.messagebus.MessageBusBackend'
+    KAFKA_GET_TIMEOUT = 0.5
+
+
+6. Configure Scrapy settings module. It's located in Scrapy project folder and referenced in scrapy.cfg. Let's add
+there::
+
+    from scrapy.settings.default_settings import SPIDER_MIDDLEWARES, DOWNLOADER_MIDDLEWARES
+
+    FRONTERA_SETTINGS = ''  # module path to your Frontera spider config module
+
+    SCHEDULER = 'frontera.contrib.scrapy.schedulers.frontier.FronteraScheduler'
+    SPIDER_MIDDLEWARES.update({
+        'frontera.contrib.scrapy.middlewares.schedulers.SchedulerSpiderMiddleware': 999,
+        'frontera.contrib.scrapy.middlewares.seeds.file.FileSeedLoader': 1,
+    })
+    DOWNLOADER_MIDDLEWARES.update({
+        'frontera.contrib.scrapy.middlewares.schedulers.SchedulerDownloaderMiddleware': 999,
+    })
+
+
+Starting the cluster
+====================
+
```

Comment:
I think there's a note needed here saying that frontera is sensitive to the order in which the components are started. First start DBW, followed by SW and than the spiders.